source/tyr/migrations/versions/04f9b89e3ef5_add_no_access_in_user_type.py

```diff
@@ -0,0 +1,47 @@
+"""Add a new type no_access in user_type
+
+Revision ID: 04f9b89e3ef5
+Revises: 75681b9c74aa
+Create Date: 2022-12-30 11:23:58.436438
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '04f9b89e3ef5'
+down_revision = '75681b9c74aa'
+
+from alembic import op
+import sqlalchemy as sa
+
+new_options = ('with_free_instances', 'without_free_instances', 'super_user', 'no_access')
+old_type = sa.Enum('with_free_instances', 'without_free_instances', 'super_user', name='user_type')
+new_type = sa.Enum(*new_options, name='user_type')
+tmp_type = sa.Enum(*new_options, name='_user_type')
+
+
+def upgrade():
+    tmp_type.create(op.get_bind(), checkfirst=False)
+    op.execute('ALTER TABLE "user" ALTER COLUMN type DROP DEFAULT')
+    op.execute('ALTER TABLE "user" ALTER COLUMN type TYPE _user_type USING type::text::_user_type')
```

Comment:
What is the purpose of this line (and line 29) ?

Reply:
Temporary user_type while adding or deleting an element in the existing user_type.